File: instruct_generations/cpp/files_with_feedback/question-5-pass-1.cpp
#include <bits/stdc++.h>
using namespace std;
int countQuadruples(const array<int, 4>& nums) {
    int count = 0;
    for (int i = 0; i < 4; i++) {
        for (int j = i + 1; j < 4; j++) {
            for (int k = j + 1; k < 4; k++) {
                for (int l = k + 1; l < 4; l++) {
                    if (nums[i] * nums[j] == nums[k] * nums[l]) {
                        count++;
                    }
                }
            }
        }
    }
    return count;
}
int main() {
    array<int, 4> nums = {1, 2, 3, 4};
    cout << countQuadruples(nums) << endl;
    return 0;
}

File: feedback_pipeline/generations/python/files_with_feedback/question-5-pass-1.py
def sameProductQuadruples(nums, N):
    count = 0
    for i in range(N):
        for j in range(i+1, N):
            for k in range(j+1, N):
                if nums[i] * nums[j] == nums[k] * nums[k+1]:
                    count += 1
    return count
